src/store/slices/authSlice.ts

```diff
@@ -0,0 +1,118 @@
+import { createAsyncThunk, createSlice, PayloadAction } from "@reduxjs/toolkit";
+import {
+	loginApi,
+	LoginResponse,
+	logoutApi,
+	registerApi,
+	RegisterResponse,
+} from "~/services/authApi";
+import { Account } from "~/types/interface/Account";
+import { AuthState } from "~/types/interface/AuthState";
+import { RegisterUser } from "~/types/interface/RegisterUser";
+import { User } from "~/types/interface/User";
+
+const initialState: AuthState = {
+	user: null,
+	accessToken: null,
+	refreshToken: null,
+	error: null,
+	loading: false,
+};
+
+export const login = createAsyncThunk<
+	LoginResponse,
+	Account,
+	{ rejectValue: string }
+>("auth/login", async (account, { rejectWithValue }) => {
+	try {
+		const response = await loginApi(account);
+		return response;
+	} catch (error: unknown) {
+		const err = error as { response?: { data?: { message?: string } } };
+		return rejectWithValue(err.response?.data?.message || "Login failed");
```

Comment:
Tương tự với những cái khác
```suggestion
		return rejectWithValue(err.response?.data?.message ?? "Login failed");
```